actionmailer/lib/action_mailer/rescuable.rb

```diff
@@ -0,0 +1,27 @@
+module ActionMailer #:nodoc:
+  # Provides `rescue_from` for mailers. Wraps mailer action processing,
+  # mail job processing, and mail delivery.
+  module Rescuable
+    extend ActiveSupport::Concern
+    include ActiveSupport::Rescuable
+
+    class_methods do
+      def handle_exception(exception) #:nodoc:
+        rescue_with_handler(exception) || raise(exception)
+      end
+    end
+
+    def handle_exceptions #:nodoc:
+      yield
+    rescue => exception
+      rescue_with_handler(exception) || raise
+    end
```

Comment:
Called by `MessageDelivery#deliver_*` to handle exceptions raised by message delivery.